### DISCUSSION TITLE
Trying to reproduce AUC with sklearn (AUC discrepancy)

### DISCUSSION THREAD

**Discussion by josp70** [pycaret/pycaret#3679] [Q&A]
![image](https://github.com/pycaret/pycaret/assets/9974954/c10a2bba-69c8-4e4b-a799-f0d566f043f1)

Hello, I'm trying to compute AUC using sklearn but I'm not able to get the same results as the AUC reported from pycaret predict_model. See for instance below, s is an experiment instance, TARGET is the target:

`pred = s.predict_model(xgb, data=data_unseen)`

![image](https://github.com/pycaret/pycaret/assets/9974954/1ca941b3-0c68-470e-ad16-42d601073626)
![image](https://github.com/pycaret/pycaret/assets/9974954/f00834d2-2a91-413c-a217-a4d697cac88a)

If I compute the AUC using sklearn I get a quite distinct value: 0.4546431657741345

```
roc_auc_score(pred.TARGET, pred.prediction_score)
```

![image](https://github.com/pycaret/pycaret/assets/9974954/3c8aabf6-80b5-43a7-a0f5-e0aa4e553752)

I should be doing something wrong, but I can not spot it.

Any idea?

**Comment by ngupta23** [comment]
Post a reproducible example (code + data) under "Issues" so someone can reproduce it and give an answer

**Comment by josp70** [comment]
Reported as issue at #3686
